### ISSUE TITLE
Brave AdBlock - Lists issue

### ISSUE THREAD

**Issue by nightznero** [brave/brave-browser#34411]
Helloz,
There is issue with `Oficjalne Polskie filtry do AdBlocka` and `Oficjalne Polskie filtry przeciwko Alertom o AdBlocku`
They are crashing `poczta.wp.pl` the site is whole white while its Email site and visinble of mails is necessary. Could You please whitelist Rules that make that page broken?
![Screenshot_20231117_104255_Brave](https://github.com/brave/brave-browser/assets/61750641/b1fc32dc-f0f3-4901-9c3b-fb10bffe2f76)

Best Regards, NIGHTZNERO.

**Comment by levicki** [comment]
@nightznero Brave probably pulls the lists from [here](https://github.com/MajkiIT/polish-ads-filter), (you can check by opening the list in Brave if possible and comparing to [this one](https://github.com/MajkiIT/polish-ads-filter/blob/master/polish-adblock-filters/adblock.txt)). Basically, you should report this issue to the adblock list maintainers, not to Brave developers.

**Comment by nightznero** [comment]
> @nightznero Brave probably pulls the Polish lists from [here](https://github.com/MajkiIT/polish-ads-filter), (you can check by opening the list in Brave if possible and comparing to [this one](https://github.com/MajkiIT/polish-ads-filter/blob/master/polish-adblock-filters/adblock.txt)). Basically, you should report this issue to the adblock list maintainers, not to Brave developers.

I report it into Brave cuz I have same list on my ADG amd its works normally but when i enable it on Brave site is broken,so i dont think it should be reported to list maintenaners but to Brave instead

**Comment by levicki** [comment]
@nightznero If you at least bothered to check the link I gave you, then you'd have seen this:

![image](https://github.com/brave/brave-browser/assets/16415478/7d6a0cb9-5b7e-4879-b351-856ae589991e)

I assume that by ADG you mean AdGuard and as you can see the AdGuard list is a separate one, and probably not the one Brave is using.

If you compare [adblock.txt](https://github.com/MajkiIT/polish-ads-filter/raw/master/polish-adblock-filters/adblock.txt) and [adblock_adguard.txt](https://raw.githubusercontent.com/MajkiIT/polish-ads-filter/master/polish-adblock-filters/adblock_adguard.txt) lists you will see that the adguard list does not have a single mention of `poczta.wp.pl`, while the regular adblock list (i.e. the one probably used by Brave has several &mdash; [1](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L4738), [2](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5021), [3](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5181), [4](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5182), [5](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5183), [6](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5184), and [7](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5185), one of them even marked as experimental.

TL;DR &mdash; issues with the adblock filter lists are not Brave's fault nor responsibility, because none of the lists are maintained by Brave.

**Comment by nightznero** [comment]
> @nightznero If you at least bothered to check the link I gave you, then you'd have seen this:
> 
> ![image](https://user-images.githubusercontent.com/16415478/284011200-7d6a0cb9-5b7e-4879-b351-856ae589991e.png)
> 
> I assume that by ADG you mean AdGuard and as you can see the AdGuard list is a separate one, and probably not the one Brave is using.
> 
> If you compare [adblock.txt](https://github.com/MajkiIT/polish-ads-filter/raw/master/polish-adblock-filters/adblock.txt) and [adblock_adguard.txt](https://raw.githubusercontent.com/MajkiIT/polish-ads-filter/master/polish-adblock-filters/adblock_adguard.txt) lists you will see that the adguard list does not have a single mention of `poczta.wp.pl`, while the regular adblock list (i.e. the one probably used by Brave has several — [1](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L4738), [2](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5021), [3](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5181), [4](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5182), [5](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5183), [6](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5184), and [7](https://github.com/MajkiIT/polish-ads-filter/blob/0808f9f98dbe5fdc68849c2eb0cb87511e2a2cea/polish-adblock-filters/adblock.txt#L5185), one of them even marked as experimental.
> 
> TL;DR — issues with the adblock filter lists are not Brave's fault nor responsibility, because none of the lists are maintained by Brave and reporting them here is a waste of both your time and Brave developers' time.
> 
> Before reporting upstream, you should try to pinpoint the exact rule which is causing the issue. In order to do that, just copy those seven rules I linked to custom filters and disable the lists, then remove the rules from custom filters one by one until the page starts working.

ADG have preinstalled list `https://raw.githubusercontent.com/MajkiIT/polish-ads-filter/master/polish-adblock-filters/adblock.txt` 
in URL i dont see _adguard so simple (if i k not wrong) its the same list handled also by Brave.
i know how rules works cuz have my own list so know how to whitelist it but if you provide instruction how to do it,can't day less thsn Thanks,its kind. Just wanted to report it cuz if the same list is handled by adg and worke fine and by Brave and have issue with adblocking normal and agressiv so dont bothered to whitelist it one by one cyz it should be a Brave issue with shields

**Comment by levicki** [comment]
If you are using both Brave and ADG at the same time then there is no need to use the same list in both, just don't enable it in Brave. Not sure if I am understanding what you are doing with ADG so I might be wrong.

**Comment by Emi-Emi-Emi** [comment]
That's not a 'crashing' that's just 'breaking' the website because some outdated or not needed rule.

You have to report it to their list maintainers, but first you need to know which rules is breaking the website, because sometimes it could be some Brave issue, and then Brave has to made a workaround it. So you first make sure, which one it is.

We have to remember most lists are made thinking about Desktop and not mobile, and sometimes it would break things because they change something between the responsiveness of the site.

To troubleshoot this, especially in Android, is kind of... complicated, I know, Brave is missing some exceptions to remove cosmetics and all that. Also the problem is Regional, Custom lists and cosmetics, the Standard vs Aggressive adblocker mode doesn't apply, so you can't set the adblocker to standard to turn the cosmetics off like with default lists.

This sounds like a simple cosmetic problem, cosmetics = the adblocker just adds `display: none !important` to an element.

Syaing this, all the limitations don't mean you can't be more specific or see the lists and trouble shoot one by one.
Like the limitation is some filters will NOT apply to lists depending where the lists is.
Brave has two adblock engines one for the default lists and the other for regionals, and custom. So some exceptions will NOT apply to default lists, in this case, being a Regional list(s) you can use custom filters without any issue to troubleshoot this and be able to make sure if it is Brave missing a feature or a bad rule that has to be made properly so it 'adapts' to mobile and desktop in case this is the reason for the breakage.

Like you can be sure which list is breaking the site, you mention two regional lists:

[Oficjalne Polskie Filtry do AdBlocka](https://github.com/brave/adblock-resources/blob/1f76ddcead370f136eb50d7450858800cacaf363/filter_lists/list_catalog.json#L899) uses only one list internally.

[Oficjalne Polskie Filtry Przeciwko Alertom o Adblocku](https://github.com/brave/adblock-resources/blob/1f76ddcead370f136eb50d7450858800cacaf363/filter_lists/list_catalog.json#L918) which uses two, but this one doesn't have any rule about poczta.wp.pl, because the rules that mention it, are commented out with the exclamation symbol `!`,  so you need to try to be more specific when reporting an issue like this.

Also, you have to remember you are the one who has access to devtools and your emails in that page and all that, some people like me will not have the same access as you do, so we can't 'test' as easy as you do, since I am not going to open an account just to help.

So the tests you can do are these ones:

First start with `@@||poczta.wp.pl^$generichide` to make sure it is not a generic cosmetic the problem.

generic = a rule that will apply to all websites. The problem is Brave doesn't support `specifichide` which will remove the website specific cometics to make sure it is that or not, because like I said, Standard mode won't apply to a regional list.

Anyway second test second `poczta.wp.pl#@#+js()` to make sure is not a scriptlet injection rule.
The two engines exceptions information I said matters here, for this second test you can use it with regional but not default lists.

Now, talking about which lists is exactly breaking the site.

### Oficjalne Polskie Filtry do AdBlocka, these are the exception cosmetics you should try if this list is breaking it, if the site is fixed, then you have to go one by one to find which one is exactly doing it.


First create an exception for page cosmetic rules, these are the only other cosmetics in the list specific to the website.

```
poczta.wp.pl#@#.ContainerBox .NativeLinkGroup
poczta.wp.pl#@#.ContainerBox native-feed[ng-if^="ListDirectiveController.currentLabel.id"]
poczta.wp.pl#@#.sidebar__outline > sidebar-feed
poczta.wp.pl#@#[slot^=BOX_LEFT_]
poczta.wp.pl#@#[slot=SKY]
```

So when you find the reason why it breaks, you report it, as you can see nothing should break anything. no advanced CSS selectors or anything.

There is a regex network request rule, but might not be the reason, but could be.

`@@/https?:\/\/(?!(poczta|bc))[a-z.]{3,15}\.wp\.pl\/.{20,}/$script,domain=poczta.wp.pl`

First test with them, because so many websites have the `wp.pl` that it could be another rule that applies to more websites.

`@@pl/*/rodzinnawycieczka/$image,domain=wp.pl`

And then you test other cosmetics:

```
~wideo.wp.pl,~www.wp.pl,wp.pl#@#A[href*=".wp.pl/"] > DIV > IMG[class][src^="https://v.wpimg.pl/"]
~www.wp.pl,wp.pl#@#DIV[style*="overflow: hidden; z-index: 310;"]
~www.wp.pl,wp.pl#@#[class^="__bc_layer"]
~www.wp.pl,~pilot.wp.pl,wp.pl#@#[class^="css-"]
~www.wp.pl,wp.pl#@#body > script + iframe + div:not([id])[class]
```

```
wp.pl#@#[href^="http://emisja.contentstream.pl/_/ctr/"]
wp.pl#@#a[href^="http://smartclick.pl/redir"]
wp.pl#@#div[class^="_"] > style + div[class]

wp.pl#@#.wp-player > .adpause > .plchldr

wp.pl#@#.scr-wallpaper
wp.pl#@#body > iframe + iframe + #pushpushgo-container + div:not([id])[class]
```

There is even one classified as 'experimental' and since it shouldn't affect the site in question, it never hurts to test it, 'just in case' no other rule fixed it.

`~pilot.wp.pl,~poczta.wp.pl,wp.pl#@#body > div:nth-of-type(1)[style^="display: block;"]:not(#page)`

The `~` symbol will create a 'don't use this domain' but as you can see it is applied to all the other `wp.pl` websites

Hope this helps to troubleshoot things. 

This is why some browser makers don't want to implement adblocker natively, a really bad excuse, but still a good excuse, too many third party lists will cause too many issues, but they should be reported to their makers unless it is a feature that Brave doesn't support, like an unsupported regex or some `$popup`  or something like that.
